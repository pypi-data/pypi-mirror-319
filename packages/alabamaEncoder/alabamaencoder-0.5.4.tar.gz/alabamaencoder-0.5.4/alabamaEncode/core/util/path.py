import os


class PathAlabama:
    def __init__(self, path: str):
        self.path = path

    def get(self):
        return self.path

    def get_safe(self):
        return f'"{self.path}"'

    def exists(self):
        return os.path.exists(self.path)

    def size_bytes(self):
        return os.path.getsize(self.path)

    def __str__(self):
        return self.get_safe()

    def check_video(self) -> bool:
        """
        Checks if the file exists and is a valid media file
        :return: True if the file exists and is a valid media file
        """
        if not self.exists():
            raise FileNotFoundError(f"File {self.path} does not exist")
        if self.size_bytes() < 10:
            raise ValueError(f"File {self.path} is too small to be a valid media file")
        return True
