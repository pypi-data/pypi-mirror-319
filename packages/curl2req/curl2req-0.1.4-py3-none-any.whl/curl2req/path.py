# -*- coding: utf-8 -*-
import json


def generate_json_paths(data, current_path="$", paths=None):
    if paths is None:
        paths = []

    if isinstance(data, dict):
        for key, value in data.items():
            new_path = f"{current_path}.{key}" if current_path != "$" else f"$.{key}"
            paths.append(new_path)
            generate_json_paths(value, new_path, paths)
    elif isinstance(data, list):
        for index, value in enumerate(data):
            new_path = f"{current_path}[{index}]"
            paths.append(new_path)
            generate_json_paths(value, new_path, paths)

    return paths


def parse_json_and_get_first_path(json_content):
    try:
        if isinstance(json_content, str):
            data = json.loads(json_content)
        else:
            data = json_content

        paths = generate_json_paths(data)

        if paths:
            return paths
        else:
            return None
    except json.JSONDecodeError:
        return "Invalid JSON content"


def get_json_paths(json_content, show=True):
    paths = parse_json_and_get_first_path(json_content)
    if show:
        for path in paths:
            print(path)
    return paths
