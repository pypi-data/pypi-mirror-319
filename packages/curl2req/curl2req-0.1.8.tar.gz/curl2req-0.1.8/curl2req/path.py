# -*- coding: utf-8 -*-
import json
import re
from typing import Union, List, Optional


def generate_json_paths(data, current_path="$", paths=None):
    """Recursively generate JSON paths for a given data structure."""
    if paths is None:
        paths = []

    if isinstance(data, dict):
        for key, value in data.items():
            new_path = f"{current_path}.{key}" if current_path != "$" else f"$.{key}"
            paths.append(new_path)
            generate_json_paths(value, new_path, paths)
    elif isinstance(data, list):
        for index, value in enumerate(data):
            new_path = f"{current_path}[{index}]"
            paths.append(new_path)
            generate_json_paths(value, new_path, paths)

    return paths


def uniq_json_paths(paths: List[str]) -> List[str]:
    """Deduplicate array paths by replacing indices with []."""
    seen = set()
    unique_paths = []

    for path in paths:
        # Replace array indices with []
        deduped_path = re.sub(r'\[\d+\]', '[]', path)
        if deduped_path not in seen:
            seen.add(deduped_path)
            unique_paths.append(deduped_path)

    return unique_paths


def parse_json_and_get_first_path(json_content):
    """Parse JSON content and generate paths."""
    try:
        if isinstance(json_content, str):
            data = json.loads(json_content)
        else:
            data = json_content

        paths = generate_json_paths(data)

        if paths:
            return paths
        else:
            return None
    except json.JSONDecodeError:
        return "Invalid JSON content"


def get_json_paths(json_content: Union[str, dict], show: bool = True, uniq_path: bool = True) -> Optional[List[str]]:
    """Get JSON paths with optional deduplication and display."""
    paths = parse_json_and_get_first_path(json_content)

    if paths is None:
        return None

    if uniq_path:
        paths = uniq_json_paths(paths)

    if show:
        for path in paths:
            print(path)

    return paths
