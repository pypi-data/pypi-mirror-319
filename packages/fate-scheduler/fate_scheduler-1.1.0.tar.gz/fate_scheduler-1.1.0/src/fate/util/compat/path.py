# added to pathlib in 3.9
def _is_relative_to(path, *other):
    """Return True if the path is relative to another path or False.
    """
    try:
        path.relative_to(*other)
        return True
    except ValueError:
        return False


def is_relative_to(path, *other):
    try:
        is_relative_to = path.is_relative_to
    except AttributeError:
        return _is_relative_to(path, *other)
    else:
        return is_relative_to(*other)


# added to pathlib in 3.9
def _readlink(path):
    target = path._accessor.readlink(path)
    return path._from_parts((target,))


def readlink(path):
    try:
        readlink = path.readlink
    except AttributeError:
        return _readlink(path)
    else:
        return readlink()
