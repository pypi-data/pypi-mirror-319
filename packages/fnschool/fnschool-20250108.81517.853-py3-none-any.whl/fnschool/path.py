import os
import sys
from pathlib import Path, PosixPath
import shutil
import getpass
import platform
import subprocess
from datetime import datetime

from appdirs import AppDirs
from fnschool.inoutput import *
from fnschool.app import *


user_name = getpass.getuser()

dirs = AppDirs(app_name, app_author)

app_dpath = Path(__file__).parent
data_dpath = app_dpath / "data"
user_config_dir = Path(dirs.user_config_dir)
user_data_dir = Path(dirs.user_data_dir)


for d in [
    user_config_dir,
    user_data_dir,
]:
    if not d.exists():
        os.makedirs(d)


def get_file_mtime(path):
    mtime = os.path.getmtime(path)
    mdatetime = datetime.fromtimestamp(mtime)
    return mdatetime


def get_file_ctime(path):
    ctime = os.path.getctime(path)
    cdatetime = datetime.fromtimestamp(ctime)
    return cdatetime


# The end.
