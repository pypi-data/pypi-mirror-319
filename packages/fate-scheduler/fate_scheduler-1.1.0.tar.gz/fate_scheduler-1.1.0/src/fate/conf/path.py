"""Inference, and environment variable specification, of relevant
filesystem paths.

"""
import enum
import functools
import operator
import os
import re
import sys
from dataclasses import dataclass
from pathlib import Path

from descriptors import cachedproperty, classonlymethod, classproperty

from fate.util.os import system_path


class PrefixProfile(enum.Flag):

    #
    # system: library was installed under a system (root-writable) path
    #
    # (if not set, library was installed under some user path.)
    #
    system = enum.auto()

    #
    # isolated: library was installed into a python virtual environment
    #           for the purpose of isolation
    #
    # (note: this flag may not be set for some virtual environments,
    # which seek only to isolate the library's requirements, but not
    # the tool itself -- e.g., pipx)
    #
    isolated = enum.auto()

    @classproperty
    def empty(cls):
        return cls(0)

    @classonlymethod
    def infer(cls, lib):
        """Compose profile flags appropriate to the installation context.

        Inference is overridden by the process environment variable:

            {LIB}_PREFIX_PROFILE={system,isolated,empty}

        The value of this variable may be a singular profile name, or a
        list of these, delimited by either "," (comma) or "|" (pipe).

        The special value `empty` may be specified to indicate the empty
        flag, (*i.e.*, non-system, non-isolated, or the "user-global"
        profile).

        Note: Any other non-existant profile names specified by the
        environment variable are ignored.

        """
        #
        # check for profile specified by environment variable
        #

        environ_spec = os.getenv(f'{lib}_PREFIX_PROFILE'.upper(), '')

        # environ-given names may be delimited by comma or pipe
        environ_names = re.findall(r'[^ ,|]+', environ_spec)

        # unrecognized environ-given names are simply ignored
        environ_profiles = [cls[name] for name in environ_names if name in cls.__members__]

        if 'empty' in environ_names:
            environ_profiles.append(cls.empty)

        if environ_profiles:
            return functools.reduce(operator.or_, environ_profiles)

        #
        # infer from installation
        #

        # module either installed under a user home directory
        # (and so will use XDG_CONFIG_HOME, etc.)
        # OR appears global
        # (and so will install global)
        location_profile = cls.system if system_path(Path(__file__)) else cls.empty

        if (
            # using system python
            sys.prefix == sys.base_prefix

            # using tox venv: treat as non-isolated
            or 'pipx' in Path(sys.prefix).parts
        ):
            return location_profile

        # looks isolated (a venv)
        # (and so will construct path from `sys.prefix`)
        return location_profile | cls.isolated


def path(method):
    """Wrap a Path-returning method such that:

    * it may be overridden by an environment variable

    * its result (if any) is given a leaf directory named for the
      PrefixPath.lib attribute

    """
    @functools.wraps(method)
    def wrapped(self, *args, **kwargs):
        if override := os.getenv(f'{self.lib}_PREFIX_{method.__name__}'.upper()):
            return Path(override).absolute()

        method_path = method(self, *args, **kwargs)

        return method_path and method_path / self.lib

    return wrapped


def path_property(prop):
    """Wrap a `Path`-returning method as a `path` and as a
    `cachedproperty`.

    See: `path`.

    """
    return cachedproperty(path(prop))

path.property = path_property


@dataclass
class PrefixPaths:
    """Path prefixes appropriate to the environment.

    Overrides to inference and defaults are retrieved from the
    process environment variables:

        {LIB}_PREFIX_PROFILE={system,isolated}

        {LIB}_PREFIX_{FIELD}=path

    """
    lib: str
    profile: PrefixProfile

    @classonlymethod
    def infer(cls, lib):
        profile = PrefixProfile.infer(lib)
        return cls(lib, profile)

    @path.property
    def conf(self):
        """library configuration"""
        if PrefixProfile.system in self.profile:
            return Path(os.sep) / 'etc'

        if PrefixProfile.isolated in self.profile:
            return Path(sys.prefix)

        if xdg_config := os.getenv('XDG_CONFIG_HOME'):
            return Path(xdg_config)

        return Path.home() / '.config'

    @path.property
    def data(self):
        """results directory (default)"""
        if PrefixProfile.system in self.profile:
            return Path(os.sep) / 'var' / 'log'

        if PrefixProfile.isolated in self.profile:
            return Path(sys.prefix)

        if xdg_data := os.getenv('XDG_DATA_HOME'):
            return Path(xdg_data)

        return Path.home() / '.local' / 'share'

    @path.property
    def state(self):
        """library (retry records) and task state"""
        if PrefixProfile.system in self.profile:
            return Path(os.sep) / 'var' / 'lib'

        if PrefixProfile.isolated in self.profile:
            return Path(sys.prefix)

        if xdg_state := os.getenv('XDG_STATE_HOME'):
            return Path(xdg_state)

        return Path.home() / '.local' / 'state'

    @path.property
    def run(self):
        """run (lock) files"""
        if PrefixProfile.system in self.profile:
            return Path(os.sep) / 'run'

        if PrefixProfile.isolated in self.profile:
            return Path(sys.prefix)

        if xdg_runtime := os.getenv('XDG_RUNTIME_DIR'):
            return Path(xdg_runtime)

        return Path.home() / '.local' / 'run'

    @path
    def completions(self, shell_name, force_system=None):
        """shell completion files"""
        dir_name = 'bash-completion' if shell_name == 'bash' else shell_name

        if force_system or (force_system is None and PrefixProfile.system in self.profile):
            return Path(os.sep) / 'usr' / 'share' / dir_name / 'completions'

        # completions must ignore user virtual env (and shouldn't matter)

        data_path = (Path(xdg_data) if (xdg_data := os.getenv('XDG_DATA_HOME'))
                     else Path.home() / '.local' / 'share')

        return data_path / dir_name / 'completions'
