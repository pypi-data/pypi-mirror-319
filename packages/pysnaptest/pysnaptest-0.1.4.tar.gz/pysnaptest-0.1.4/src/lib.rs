use std::path::PathBuf;

use pyo3::{
    exceptions::PyValueError,
    pyclass, pyfunction, pymethods, pymodule,
    types::{PyModule, PyModuleMethods},
    wrap_pyfunction, Bound, PyAny, PyErr, PyResult,
};

const PYSNAPSHOT_SUFFIX: &str = "pysnap";

#[pyclass(frozen)]
struct TestInfo {
    test_path: PathBuf,
    test_name: String,

    snapshot_path_override: Option<PathBuf>,
    snapshot_name_override: Option<String>,
}

#[pymethods]
impl TestInfo {
    #[new]
    #[pyo3(signature = (test_name, test_path, snapshot_path_override = None, snapshot_name_override = None))]
    fn new(
        test_name: String,
        test_path: PathBuf,
        snapshot_path_override: Option<PathBuf>,
        snapshot_name_override: Option<String>,
    ) -> PyResult<Self> {
        Ok(TestInfo {
            test_name,
            test_path,
            snapshot_path_override,
            snapshot_name_override,
        })
    }

    fn snapshot_path(&self) -> PyResult<PathBuf> {
        if let Some(snapshot_path) = self.snapshot_path_override.clone() {
            return Ok(snapshot_path);
        }

        let mut test_file_dir = self
            .test_path
            .parent()
            .ok_or_else(|| {
                PyValueError::new_err("Invalid 'current_test' value - should contain a single '::'")
            })?
            .to_path_buf();

        test_file_dir.push("snapshots");

        Ok(test_file_dir)
    }

    fn snapshot_name(&self) -> String {
        if let Some(sno) = self.snapshot_name_override.as_ref() {
            return sno.clone();
        }

        let test_name = self
            .test_name
            .strip_suffix(" (call)")
            .unwrap_or(self.test_name.as_ref());
        let file_name = self.test_path.file_stem().and_then(|s| s.to_str());
        if let Some(f) = file_name {
            format!("{f}_{test_name}")
        } else {
            test_name.to_string()
        }
    }
}

impl TryInto<insta::Settings> for &TestInfo {
    type Error = PyErr;

    fn try_into(self) -> PyResult<insta::Settings> {
        let mut settings = insta::Settings::clone_current();
        settings.set_snapshot_path(self.snapshot_path()?);
        settings.set_snapshot_suffix(PYSNAPSHOT_SUFFIX);
        settings.set_input_file(&self.test_path);
        settings.set_omit_expression(true);
        Ok(settings)
    }
}

#[pyfunction]
fn assert_json_snapshot(test_info: &TestInfo, result: &Bound<'_, PyAny>) -> PyResult<()> {
    let res: serde_json::Value = pythonize::depythonize(result).unwrap();
    let snapshot_name = test_info.snapshot_name();
    let settings: insta::Settings = test_info.try_into()?;
    settings.bind(|| {
        insta::assert_json_snapshot!(snapshot_name, res);
    });
    Ok(())
}

#[pyfunction]
fn assert_csv_snapshot(test_info: &TestInfo, result: &Bound<'_, PyAny>) -> PyResult<()> {
    let res: serde_json::Value = pythonize::depythonize(result).unwrap();
    let snapshot_name = test_info.snapshot_name();
    let settings: insta::Settings = test_info.try_into()?;
    settings.bind(|| {
        insta::assert_csv_snapshot!(snapshot_name, res);
    });
    Ok(())
}

#[pyfunction]
fn assert_snapshot(test_info: &TestInfo, result: &Bound<'_, PyAny>) -> PyResult<()> {
    let snapshot_name = test_info.snapshot_name();
    let settings: insta::Settings = test_info.try_into()?;
    settings.bind(|| {
        insta::assert_snapshot!(snapshot_name, result);
    });
    Ok(())
}

#[pymodule]
#[pyo3(name = "_pysnaptest")]
fn pysnaptest(m: &Bound<'_, PyModule>) -> PyResult<()> {
    m.add_class::<TestInfo>()?;

    m.add_function(wrap_pyfunction!(assert_snapshot, m)?)?;
    m.add_function(wrap_pyfunction!(assert_json_snapshot, m)?)?;
    m.add_function(wrap_pyfunction!(assert_csv_snapshot, m)?)?;
    Ok(())
}
