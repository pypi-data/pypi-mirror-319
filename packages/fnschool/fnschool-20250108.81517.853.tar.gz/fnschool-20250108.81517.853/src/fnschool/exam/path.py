import os
import sys
from fnschool import *

exam_dpath = Path(__file__).parent
exam_data_dpath = exam_dpath / "data"
user_exam_dpath = user_data_dir / _("exam")
teach_name_fpath = user_exam_dpath / (_("teacher_name") + ".txt")
score_fpath0 = exam_data_dpath / "score.xlsx"
parental_emails_fpath0 = exam_data_dpath / "parental_emails.xlsx"

for d in [
    user_exam_dpath,
]:
    if not d.exists():
        os.makedirs(d.as_posix(), exist_ok=True)

if not teach_name_fpath.exists():
    if not teach_name_fpath.parent.exists():
        os.makedirs(teach_name_fpath.parent.as_posix(), exist_ok=True)
    with open(teach_name_fpath, "w", encoding="utf-8") as f:
        f.write("")

# The end.
