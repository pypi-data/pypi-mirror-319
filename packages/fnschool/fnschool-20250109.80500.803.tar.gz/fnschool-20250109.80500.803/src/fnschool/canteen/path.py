import os
import sys
import tomllib
import shutil
from fnschool import *
from fnschool.external import *


food_classes_config0_fpath = Path(__file__).parent / "food_classes.toml"
canteen_data_dpath = Path(__file__).parent / "data"
bill0_fpath = canteen_data_dpath / "bill.xlsx"
pre_consuming0_fpath = canteen_data_dpath / "consuming.xlsx"
user_canteen_dpath = user_data_dir / _("canteen")
operator_name_fpath = user_canteen_dpath / (_("operator_name") + ".txt")
documents_dpath = Path.home() / "Documents"

for d in [
    user_canteen_dpath,
]:
    if not d.exists():
        os.makedirs(d, exist_ok=True)

if not operator_name_fpath.exists():
    with open(operator_name_fpath, "w", encoding="utf-8") as f:
        f.write("")


# The end.
