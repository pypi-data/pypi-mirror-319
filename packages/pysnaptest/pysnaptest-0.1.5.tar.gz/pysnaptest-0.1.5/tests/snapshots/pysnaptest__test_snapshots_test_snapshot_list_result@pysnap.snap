---
source: src/lib.rs
input_file: tests/test_snapshots.py
---
[
  1,
  2,
  4
]
